File: seohwee/week1/A002.py
class Solution(object):
    def generate(self, numRows):
        array = [[1]]
        triangle = [1, 1]
        
        for i in range(numRows-1):
            array.append(triangle)
            s = [1]
            
            for j in range(len(triangle)-1):
                s.append(triangle[j] + triangle[j+1])
            s.append(1)
            triangle = s
        return array

File: SeongKyungKim/week1/A002.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        
	        List<List<Integer>> pascal = new ArrayList<>(numRows);
	        int temp;
	        
	        for(int i=0; i<numRows; i++) {
	            List<Integer> list  = new ArrayList<>(i+1);
	            list.add(1);
	           
	            for(int j=1; j<i; j++){
	                temp = pascal.get(i-1).get(j-1) + pascal.get(i-1).get(j);
	                list.add(temp);
	            }
	            
                if(i>=1) list.add(1);
	            pascal.add(list);
	        }
	    
        return pascal;
    }
}
